### PR TITLE
Resolve a few PHPstan level 7 issues

### PR DESCRIPTION
**Pull request by AJenbo**
Type: bugfix|documentation  
Breaking change: no

Gets us from 58 to 48 remaining issues at level 7. This is mainly done by checking for invalid configurations and adjusting a few signatures.

Note: The two failing tests do not appear to be related to any of my changes here.

### CODE REVIEW

File: src/main/php/PDepend/Report/Overview/Pyramid.php [hunk 1]
```diff
@@ -198,7 +198,12 @@ public function close()
         $proportions = $this->computeProportions($metrics);
 
         $svg = new DOMDocument('1.0', 'UTF-8');
-        $svg->loadXML(file_get_contents(dirname(__FILE__) . '/pyramid.svg'));
+        $templatePath = dirname(__FILE__) . '/pyramid.svg';
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Overview/Pyramid.php:201]:
```suggestion
        $templatePath = __DIR__ . '/pyramid.svg';
```
Twice the same modification, this sounds like an opportunity to dedupe the SVG loading. A `private loadSvgFile()` would be nice.

File: src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php [hunk 2]
```diff
@@ -577,10 +577,12 @@ protected function parseUseDeclarationVersion70(array $fragments)
...
 
             // Add mapping between image and qualified name to symbol table
-            $this->useSymbolTable->add($image, join('', array_merge($fragments, $subFragments)));
+            if ($image) {
+                $this->useSymbolTable->add($image, join('', array_merge($fragments, $subFragments)));
+            }
         } while (true);
 
-        if (isset($image, $subFragments)) {
+        if (isset($image, $subFragments) && $image) {
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:585]: I think we have some `$image` being `""`

Comment (kylekatarnls) [reply src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php:585]:
```suggestion
        if (isset($image, $subFragments) && $image !== false) {
```

**Review by kylekatarnls — CHANGES_REQUESTED** [review]
All good aside the open comments

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 3]
```diff
@@ -155,7 +156,12 @@ public function close()
         $bias = 0.1;
 
         $svg = new DOMDocument('1.0', 'UTF-8');
-        $svg->loadXML(file_get_contents(dirname(__FILE__) . '/chart.svg'));
+        $templatePath = dirname(__FILE__) . '/chart.svg';
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply src/main/php/PDepend/Report/Jdepend/Chart.php:159]: Oh wow you are right, I thought it wasn't added before 5.4 so I have been leaving them alone, let me just change that (hopefully this won't escalate)

File: src/main/php/PDepend/Report/Overview/Pyramid.php (hunk 1 above)
Comment (AJenbo) [reply src/main/php/PDepend/Report/Overview/Pyramid.php:201]: They are not in the same file so private at the very least is not an option. Any chance we can take this opportunity another time?

File: src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php (hunk 2 above)
Comment (AJenbo) [reply src/main/php/PDepend/Source/Language/PHP/PHPParserVersion70.php]: Done

**Review by kylekatarnls — DISMISSED** [review]

File: src/test/resources/files/bugs/030/testCorrectClassTokensAreSet.php [hunk 4]
```diff
@@ -10,7 +10,7 @@ abstract class PHP_Depend_ParserTest extends PHP_Depend_AbstractTest
      */
     public function testParserBacktickExpressionBug15()
     {
-        $sourceFile = dirname(__FILE__) . '/my/test/file.php';
+        $sourceFile = __DIR__ . '/my/test/file.php';
```
Comment (kylekatarnls) [inline src/test/resources/files/bugs/030/testCorrectClassTokensAreSet.php:13]:
```suggestion
        $sourceFile = dirname(__FILE__) . '/my/test/file.php';
```

Comment (kylekatarnls) [reply src/test/resources/files/bugs/030/testCorrectClassTokensAreSet.php:13]: It's a test file example and `testCorrectClassTokensAreSet` check token list from it, so we should let it as is.

**Review by kylekatarnls — APPROVED** [review]

File: src/test/resources/files/bugs/030/testCorrectClassTokensAreSet.php (hunk 4 above)
Comment (AJenbo) [reply src/test/resources/files/bugs/030/testCorrectClassTokensAreSet.php]: Thanks

**Comment by AJenbo** [comment]
@kylekatarnls Hm I had expected the Coverage / PHP 8.3 to pass once merged but it looks like it's still complaining about the config, any ideas?

**Comment by kylekatarnls** [comment]
At least the generate-phar pass once merged.

On this particular job the script fixing compatibility didn't find the files:
image

**Comment by kylekatarnls** [comment]
Ah oops, found. What I did to split main and test dependencies is not applied on the coverage build.

**Comment by kylekatarnls** [comment]
Fixed #715

**Pull request merged**